src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/PiglinListener.java

```diff
@@ -0,0 +1,56 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Item;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityPickupItemEvent;
+import org.bukkit.event.entity.EntityTargetEvent;
+import org.bukkit.event.player.PlayerInteractAtEntityEvent;
+import org.bukkit.event.player.PlayerInteractEntityEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+
+/**
+ * Listens to "Piglin trading events" to prevent trading with Slimefun items.
+ *
+ * @author poma123
+ */
+public class PiglinListener implements Listener {
+
+    public PiglinListener(SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onEntityPickup(EntityPickupItemEvent e) {
+        if (e.getEntityType() == EntityType.PIGLIN) {
+            if (SlimefunItem.getByItem(e.getItem().getItemStack()) != null) {
+                e.setCancelled(true);
+            }
+        }
+    }
+
+    @EventHandler
+    public void onInteractEntity(PlayerInteractEntityEvent e) {
+        if (!e.getRightClicked().isValid() || e.getRightClicked().getType() != EntityType.PIGLIN) {
+            return;
+        }
+
+        ItemStack itemStack;
+
+        if (e.getHand() == EquipmentSlot.OFF_HAND) {
+            itemStack = e.getPlayer().getInventory().getItemInOffHand();
+        } else {
+            itemStack = e.getPlayer().getInventory().getItemInMainHand();
+        }
+
```

Comment:
Also just a small performance optimization: We should check if the item is a gold ingot, that way we won't have to do any checks that are not needed.